### ISSUE TITLE
Contour line layer

### ISSUE THREAD

**Issue by Fizzie41** [openstreetmap/iD#8201]
Apologies if this has been discussed previously, but couldn't find anything for it?

Have been doing some mapping in a country area recently, using the default "best" imagery (in this case "DCS NSW Imagery":
https://www.openstreetmap.org/edit#map=18/-28.17951/152.74558

Looking at the available Background Settings, I noticed OpenTopoMap which gives topo information:
https://www.openstreetmap.org/edit#map=18/-28.17951/152.74558

Is there any way of displaying the topo information eg contour lines, over the top of the normal aerial imagery, possibly as an extra layer that can be turned on as required?

This would often be very helpful, especially in country areas where reference to high & low ground is often the best way of telling exactly where you are! :-)

Thanks

**Comment by tastrax** [comment]
This would be great as an overlay if suitable country/state datasets were available in the same way as the imagery. Can also help with river/stream flow direction.

**Comment by quincylvania** [comment]
@mikelmaron Would a contour-only overlay be possible with Mapbox?

**Comment by bhousel** [comment]
We added Stamen Terrain for this years ago:  
https://github.com/osmlab/editor-layer-index/issues/143
https://github.com/osmlab/editor-layer-index/issues/176

From what I understand Mapbox Terrain uses elevation sources that don't allow it for sharing in this way.  @ajashton knows more.

**Comment by Fizzie41** [comment]
> We added Stamen Terrain for this years ago:

Thanks, Bryan, but just the hill shading by itself doesn't do the same job as having actual contour lines to indicate the high & low ground.

**Comment by bhousel** [comment]
> Thanks, Bryan, but just the hill shading by itself doesn't do the same job as having actual contour lines to indicate the high & low ground.

What about the OpenTopoMap layer ?

<img width="963" alt="Screen Shot 2020-11-16 at 4 41 14 PM" src="https://user-images.githubusercontent.com/38784/99311430-99436380-282a-11eb-8fcc-5d52bbd86061.png">

Update:  oh I see you were asking to display that on top of other imagery.. sorry.

**Comment by Fizzie41** [comment]
Yep, that's it :-)

The combo of the aerial imagery so you can see where the road runs & where the heavy vegetation is v open grass, together with contour lines so you can say that that is the high ground up there, while the stream runs down this gully & curves around down there, so I'm here, would be great!

**Comment by RAytoun** [comment]
Contour lines are a great asset for mapping. It allows you to see the valleys and probable watercourses even under a dense canopy of trees. In JOSM I put the OpenCycleMap above my Imagery layer then I reduce the opacity of the OpenCycleMap to about 50% so I can see through it and increase the Gamma to darken the contours so that i can see the them over the imagery. I can then work out direction of flow and continuity of waterway where it is obscured by cloud or vegetation, In wet regions I can concentrate my search for settlements on the obvious higher ground in rural areas. I can identify where the cliff faces are that would present obstacles or indicate waterfalls to name a few. Having something like this in iD would be a great move forward in improving the way we make informed decisions regarding features.

**Comment by mikelmaron** [comment]
Something [like this](https://api.mapbox.com/styles/v1/mikelmaron/ckhmstnt004z419pck2yfh4yx.html?fresh=true&title=view&access_token=pk.eyJ1IjoibWlrZWxtYXJvbiIsImEiOiJjaWZlY25lZGQ2cTJjc2trbmdiZDdjYjllIn0.Wx1n0X7aeCQyDTnK6_mrGw#12.63/36.25618/-105.30263) @quincylvania ?

This is a quick iteration without much design thinking, just to illustrate it's possible. Note that terrain data only available at zoom 11 and later. If someone wanted to experiment with this, happy to provide pointers in Mapbox Studio, and when you're happy with the result, can transfer into the openstreetmap account for use in iD

**Comment by Fizzie41** [comment]
Just wondering if we were able to get anywhere with this? Thanks!

**Comment by quincylvania** [comment]
@Fizzie41 Any updates would be posted here, so don't worry about missing something. Open source development takes time.

@mikelmaron Something like that should work. Do you know if the license is compatible?

**Comment by mikelmaron** [comment]
Yes @quincylvania just like the other Mapbox layers made available in iD, this is usable.

**Comment by Fizzie41** [comment]
> Any updates would be posted here, so don't worry about missing something. Open source development takes time.

Thanks! Yes, I know that, I'm just being selfish & shuffling it back up to the top of the pile :-)

**Comment by tordans** [comment]
FYI here is an experiment with elevation data https://github.com/openstreetmap/iD/issues/12660

**Comment by Fizzie41** [comment]
Thanks!
